apps/judicial-system/backend/src/app/modules/case/guards/mergedCaseExists.guard.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  BadRequestException,
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common'
+
+import { CaseService } from '../case.service'
+import { Case } from '../models/case.model'
+
+@Injectable()
+export class MergedCaseExistsGuard implements CanActivate {
+  constructor(private readonly caseService: CaseService) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest()
+
+    const theCase: Case = request.case
+
+    if (!theCase) {
+      throw new InternalServerErrorException('Missing case')
+    }
+
+    const mergedCaseId = request.params.mergedCaseId
+
+    if (!mergedCaseId) {
+      return true
```

Comment:
Should this not be `return false`?

Reply:
No, this guard only applies if there is a `mergeCaseId`. The idea here is that if we provide a `mergeCaseId` then we replace the case with the merged case to allow us to access it in the relevant endpoint. Basically, `caseId` is used for access control to determine if the user should really be allowed to access the case. `mergeCaseId` on the other hand refers to the case we are actually accessing.

Example:
Case B is merged into case A. Case B is now closed and case A has a reference to it.
If I am the prosecutor assigned to case A, but case B came from a different prosecutors office, then I still need to be able to access case B. This guard gives me that access.
If I am a defender or spokesperson assigned to case A, but not case B, then I still need to be able to access case B. This guard gives me that access.

I hope this explaines.

Reply:
Ah, got it! Thanks